odoo/addons/base/models/res_config.py

```diff
@@ -570,10 +571,10 @@ def set_values(self):
                     value = self[name].ids
             else:
                 value = self[name]
-            IrDefault.set(model, field, value)
+            if value != current_settings[name]:
```

Comment:
`current_settings` is the dict of defaults.  Technically, you have no guarantee that `name in current_settings`.

Reply:
It's how the custom default_get behavior of res.config.settings model work, but yes, I should cover the case when name is not in current_settings :+1: